corehq/ex-submodules/casexml/apps/case/tests/util.py

```diff
@@ -19,6 +19,30 @@
 from casexml.apps.case.util import post_case_blocks
 
 
+class RestoreCaseBlock(object):
+    """
+    Little shim class for working with XML case blocks in a restore payload
+    """
+
+    def __init__(self, xml_element, version=V2):
+        self.xml_element = xml_element
+        self.version = version
+
+    def to_string(self):
+        return ElementTree.tostring(self.xml_element)
+
+    def get_case_id(self):
+        if self.version == V1:
+            return self.xml_element.findtext('{{{0}}}case_id'.format(get_case_xmlns(self.version)))
+        else:
+            return self.xml_element.get('case_id')
+
+    def get_case_name(self):
+        assert self.version == V2, 'get_case_name not yet supported for legacy V1 casexml'
+        # note: there has to be a better way to work with namespaced xpath.... right?!?!
```

Comment:
yea it's pretty horrific. i had to dive into a little bit when writing a script. you can see how app_manager does it? https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/xform.py#L27. @snopoke would definitely know more